src/replication.c

```diff
@@ -1606,32 +1618,47 @@ accept_mgmt_conns(int epfd, int sfd)
 #ifdef	DEBUG
 		replica->replica_mgmt_dport = atoi(sbuf);
 #endif
+
 		rc = epoll_ctl(epfd, EPOLL_CTL_ADD, mgmt_fd, &event);
 		if(rc == -1) {
 			REPLICA_ERRLOG("epoll_ctl() failed on fd(%d), "
 			    "err(%d).. closing it.. for replica(%s:%d)\n",
 			    mgmt_fd, errno, replica->ip, replica->port);
 cleanup:
+			if (mevent1) {
```

Comment:
lets delete mevent1 and mevent2 after doing EPOLL_CTL_DEL as the pointers are added to epoll.